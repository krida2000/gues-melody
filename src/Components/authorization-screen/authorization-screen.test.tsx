import * as React from "react";
import * as renderer from "react-test-renderer";
import AuthorizationScreen from "./authorization-screen";
import {BrowserRouter} from "react-router-dom"
const noop = () => {
  // do nothing
};
describe(`Authorization screen correctly render`, () => {
  it(`if authorization required`, () => {
    const tree = renderer
      .create(
        <AuthorizationScreen
          isAuthorizationRequired={true}
          onAuthorization={noop}
        />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`if authorization don't required`, () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AuthorizationScreen
            isAuthorizationRequired={false}
            onAuthorization={noop}
          />
        </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
})
