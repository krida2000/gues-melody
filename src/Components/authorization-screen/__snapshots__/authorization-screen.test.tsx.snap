// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization screen correctly render if authorization don't required 1`] = `null`;

exports[`Authorization screen correctly render if authorization required 1`] = `
<form
  action=""
  className="login__form"
>
  <p
    className="login__field"
  >
    <label
      className="login__label"
      htmlFor="name"
    >
      Логин
    </label>
    <input
      className="login__input"
      id="name"
      name="name"
      type="text"
    />
  </p>
  <p
    className="login__field"
  >
    <label
      className="login__label"
      htmlFor="password"
    >
      Пароль
    </label>
    <input
      className="login__input"
      id="password"
      name="password"
      type="password"
    />
    <span
      className="login__error"
    >
      Неверный пароль
    </span>
  </p>
  <button
    className="login__button button"
    onClick={[Function]}
    type="submit"
  >
    Войти
  </button>
</form>
`;
